middleware/kubernetes/k8sclient/k8sclient.go

```diff
@@ -23,13 +25,17 @@ type K8sConnector struct {
 }
 
 func (c *K8sConnector) SetBaseUrl(u string) error {
-    validUrl, error := url.Parse(u)
+    url, error := url.Parse(u)
 
     if error != nil {
         return error
     }
-    c.baseUrl = validUrl.String()
 
+	if ! url.IsAbs() {
+		return errors.New("k8sclient: Kubernetes endpoint url must be an absolute URL")
+	}
+
+    c.baseUrl = url.String()
```

Comment:
Why is baseUrl a string and not the url object? Also we should probably use baseURL (govet will tell you about that)